### ISSUE TITLE
No warning for unused import in `__init__.py`

### ISSUE THREAD

**Issue by rrthomas** [pylint-dev/pylint#10162]
### Bug description

In a directory `tmp` I put the following code as `__init__.py`:

```python
"""Test module"""
import sys
```

Pylint 3.3.3 reports no problems with this code:

```sh
$ pylint __init__.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

This is odd, as I have an unused import.

If I copy the same code to `test.py` and rerun pylint, I get the expected error:

```sh
$ pylint test.py
************* Module tmp.test
test.py:2:0: W0611: Unused import sys (unused-import)

-----------------------------------
Your code has been rated at 0.00/10
```

### Configuration

```ini

```

### Command used

```shell
pylint test.py
```

### Pylint output

```sh
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

```sh
__init___.py:2:0: W0611: Unused import sys (unused-import)

-----------------------------------
Your code has been rated at 0.00/10
```

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0]
```

### OS / Environment

Ubuntu 24.04

### Additional dependencies

```python

```

**Comment by zenlyj** [comment]
by default `unused-import` checks are skipped for `__init__.py` files, but you may enable the check using the `init-import` configuration.
```
pylint --init-import yes __init__.py
```
alternatively specify in your pylintrc file:
```
[VARIABLES]
init-import=yes  # this is "no" by default
```

**Comment by rrthomas** [comment]
Thanks, it never occurred to me that this would be deliberate behaviour! I found the documentation for the flag, but for those of us whose primary interaction with Pylint is via errors in our IDE (in my case, VS Codium), it would be nice if it could be documented in the warning's page (I checked https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unused-import.html and found no mention), along with a rationale (I don't understand why this happens, it has merely confused and then annoyed me!).

**Comment by rrthomas** [comment]
Much appreciated, many thanks! I'm sure future me, having forgotten all of this, will benefit from the documentation.
